Added support for sub transports

jrgkraus opened the pull request:
When "include subtransports" is active, a new column "main transport" appears showing the main transport where the actual task belongs to:
![2025-04-14 15_12_45-ZABAP Transport of Copies](https://github.com/user-attachments/assets/14356905-91c4-465f-bd4b-44d42bc5646c)

The program adds sort criteria in order to show the the work load in the same sequence as SE01 does in the tree view. Therefore, an additional field CREATE_DATE_MAIN has been added to the main table.

Furthermore, a column showing if the task / order  is already released appears in leftmost position

After having created a ToC, a double click on its number will now jump to SE01 showing the newly created transport

jrgkraus committed:
Add more support for sub transports

When working with sub transports, it is helpful to see also the main transport in the line. 
This fork adds a new column "main transport" where the upper transport number will be displayed
When "sub transports" is deactivated, the column is hidden
Furthermore:
- add an icon in front of each line that indicates the release state of the order
- add support for double click on the created ToC order - same reaction as clicking on the transport order

jrgkraus committed:
Update README.md

Kaszub09 commented:
Hi, thanks for new functions. There are few thing that need to be changed before I can merge.

1. Function `setup_sorts` is unnecessary. I don't think any sorts should be hardcoded, that's what the layouts are for, and can be saved as initial if you prefer that sort on your system.
2. Function `provide_main_transport` is not needed, you may as well do `case` inside query. Additionally, as far as I know  the whole `value #( for line in lines...` code will effectively make 2 copies per line (one with `line` and another with `value # (base line`, and another copy for assigning result, where you could simply `LOOP AT ... INTO REFERENCE...` and change it directly without copying anything.
3. Exact same comment goes for `set_sort_date`, could be done using `case` in query, no need for searching table again - plus if you searched by `lines[ transport = line-main_transport ]`, there should be a secondary sorted key to speed up search.
4. A simple check if ToC number is not initial on double click would be nice.
5. Why is there CDS that isn't used? zds_abap_toc_transports
6. The whole try/catch and raising of `cx_salv_error` should be removed. I see no point in basically bloating the code with it and raising declarations, since realistically speaking it should never occur (e.g. you set column text, it should always exists - I think it's even preferable to fail with dump so you notice immediately in DEV because 99.9% it's a code error). Plus if you have report and `cl_salv_table` fails there isn't much you can do.
7. Finally, it would be nice if formatting was more consistent with existing code, especially uppercase for keywords, parameters names, but possibly methods and data declarations as well. 
Current:
 ![obraz](https://github.com/user-attachments/assets/514f735f-1109-4b78-8f41-baaac021be14)
Preferred: (methods with more arguments could be broken into several lines)
![obraz](https://github.com/user-attachments/assets/28a653f6-c42f-4da7-a704-8582caaeb5f2)

jrgkraus committed:
Delete src directory

jrgkraus committed:
Create dummy

jrgkraus committed:
Adhere to initial programming style

The coding has been adapted to the initial programming style. Some functionality has been moved from methods to the SQL query.

jrgkraus committed:
Update ztoc.prog.abap

remove try/catch

jrgkraus committed:
Update zcl_zabap_toc_report.clas.abap

remove raising

jrgkraus commented:
Hi, thanks for reviewing. Sorry that I did not take care of the initial programming style. The CDS was remaining trash from a further tentative, So i removed it. 
Good point regarding PROVIDE_MAIN_TRANSPORT. I actually tried to do it with a CASE but did not succeed. Now, I pushed myself deeper in and got it to work. Same for SET_SORT_DATE
I removed all try/catch on CX_SALV... although, in my own projects, I prepare to not have warnings from the syntax check. But this is again a personal preference. 
Regarding SETUP_SORTS.. I actually did not remove it. The list is designed to have this kind of sort in order to not let appear the main contract multiple times in the rows, so in my opinion, providing the corresponding sort oder by default makes part of the implementation. What do you think?
